server/map_rpc_server.go

```diff
@@ -160,18 +160,17 @@ func (t *TrillianMapServer) GetLeavesByRevisionNoProof(ctx context.Context, req
 	if req.Revision < 0 {
 		return nil, fmt.Errorf("map revision %d must be >= 0", req.Revision)
 	}
-	if err := hasDuplicates(req.Index); err != nil {
-		return nil, err
-	}
 	tree, hasher, err := t.getTreeAndHasher(ctx, req.MapId, optsMapRead)
 	if err != nil {
 		return nil, fmt.Errorf("could not get map %v: %v", req.MapId, err)
 	}
+	indexValidator := newRequestIndexValidator(hasher.Size())
```

Comment:
This object is stateful, and its entire state (lots of keys potentially) will stick around in memory while the transaction is ongoing. How about moving it to a function, for better GC-ability?
Also, when I see a function name like "validate", I think of something stateless.

Based on these 2 arguments, how about just having a simple `validateLeafIndices` function instead of a dedicated type?

Reply:
Oh I see that the 2 things loop over different types of arrays. That's a bit sad then...

Reply:
Turned into a single method that sucks in the iteration and thus avoids being stateful.